renku/cli/dataset.py

```diff
@@ -297,71 +292,6 @@ def prompt_tag_selection(tags):
     return None
 
 
-def download_file_with_progress(extract, data_folder, file, chunk_size=16384):
```

Comment:
This method was put here when we refactored renku into `cli` and `core`. The download method was originally part of the download code, but since it contains cli/ui stuff (showing the progress bar via `tqdm`) that doesn't make sense in e.g. the Service part, I moved the download method to `renku.cli`. Ideally, only the progress bar parts would have to be injected and the main download functionality would be in `renku.core` but due to the download code being a bit of a mess, I decided that would be too much effort to split up and it'd be easier to inject download as a single function (plus multiprocessing also doesn't make things any easier). See https://github.com/SwissDataScienceCenter/renku-python/issues/737 and https://github.com/SwissDataScienceCenter/renku-python/issues/737 

Moving tqdm back to `renku.core` is suboptimal, but i guess using the `disabled=` property works. The main issue is, if the UI ever implements polling for this operation or or there is some other client that needs feedback on the progress of the operation in some other form, this will be a mess. Now that we're actually implementing the renku service, we can make a more informed decision than we did back then, maybe @jsam can chime in on this.

Changing import to use dataset add is a neat idea, though.

Reply:
I've added a callback class and moved the progress bar out of the core.

Reply:
Great, thank you!